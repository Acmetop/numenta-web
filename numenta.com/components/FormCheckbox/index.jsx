import React from 'react'

import styles from './index.css'


/**
 *
 */
const FormCheckbox = ({name}) => (
  <input
    className={styles.checkbox}
    id={name}
    name={name}
    type="checkbox"
  />
)

FormCheckbox.propTypes = {
  name: React.PropTypes.string.isRequired,
}

export default FormCheckbox
