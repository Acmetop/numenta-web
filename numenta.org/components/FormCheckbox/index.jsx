// Numenta.org HTM Community website source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const FormCheckbox = ({name, onClick}) => (
  <input
    className={styles.checkbox}
    id={name}
    name={name}
    onClick={onClick}
    type="checkbox"
  />
)

FormCheckbox.propTypes = {
  name: React.PropTypes.string.isRequired,
  onClick: React.PropTypes.func,
}

export default FormCheckbox
