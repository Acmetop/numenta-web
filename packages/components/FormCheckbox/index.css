/**
 * Numenta shared web components
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.checkbox {
  composes: mr1 from 'tachyons';
  composes: relative from 'tachyons';
  top: -0.1rem;
}
