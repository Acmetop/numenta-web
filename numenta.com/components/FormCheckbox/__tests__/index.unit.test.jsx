import React from 'react'
import renderer from 'react-test-renderer'

import FormCheckbox from '../../FormCheckbox'


describe('<FormCheckbox /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <FormCheckbox name="field" />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
