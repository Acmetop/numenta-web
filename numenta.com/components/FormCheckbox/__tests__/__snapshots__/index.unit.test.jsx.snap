exports[`<FormCheckbox /> React component unit test suite Renders correctly 1`] = `
<input
  className="checkbox"
  id="field"
  name="field"
  onClick={undefined}
  type="checkbox" />
`;
